### ISSUE TITLE
"Alpha ESS: Set Battery Charge" not setting

### ISSUE THREAD

**Issue by braindeadmalc** [CharlesGillanders/homeassistant-alphaESS#138]
Automations using "Alpha ESS: Set Battery Charge" don't set the battery charge period on my Smile B3. 
No errors, just not setting the times

These automations worked prior to the latest Alpha upgrade.

Looking at the Alpha API site, they appear to be stating  that setting one per day is allowed which is what caused problems for a lot of us. It got changed to once every 15 mins but I'm wondering if they've put it back to once per day?

**Comment by mapo-89** [comment]
I have the same problem, as well as with the discharge

**Comment by dan22h** [comment]
Same problem on my Smile5 but it doesn't even work once per day so I don't imagine that's the issue. I really need to convert to a direct MODBUS connection but I can't get it to work on my system for some unknown reason.

**Comment by mapo-89** [comment]
But it doesn't work with Postman either. So the problem seems to be with the API

**Comment by Euan-p** [comment]
My SMILE5 isn't working either, but only the charge setting for some reason? Is throwing up an error of "unknown action".

**Comment by DavidqStokes** [comment]
Just done some testing via my scheduled automation.
Although HA doesn't appear to be throwing any errors into the logs, the update to the SMILE B3 settings fails (as confirmed via the AlphaESS app)

**Comment by Poshy163** [comment]
Cheers for the reports guys, I'm working on a new PR to hopefully improve the charging/discharging stuff. Great to have alpha throw some edge cases these last few weeks so I can verify if the whole system works 😅

**Comment by DavidqStokes** [comment]
> Cheers for the reports guys, I'm working on a new PR to hopefully improve the charging/discharging stuff. Great to have alpha throw some edge cases these last few weeks so I can verify if the whole system works 😅

And the on-going work on this (by everyone involved) is appreciated as always.

**Comment by braindeadmalc** [comment]
I've raised a ticket with Alpha assuming it's their issue.

**Comment by braindeadmalc** [comment]
Here's the reply from Alpha

Description Customer message ince no rs/h The Onen APL "Set Charging Information" called from the t long Com prob ome alphi 2424 Com Cu View Andrew Bailey 2024-10-23 08:14:46 
Dear Customer, Thank you for reaching out to us and bringing this to our attention. We are working on this issue and expected to take few days to resolve. If you still experience the issue on 28/10, please create a new support request again. Sorry for any inconvenience it may caused.
 Kind Regards, Eddie Yam Alpha ESS UK Evaluate

**Comment by braindeadmalc** [comment]
Here's the reply from Alpha 

We are working on this issue and expected to take few days to resolve. If you still experience the issue on 28/10, please create a new support request again. Sorry for any inconvenience it may caused. Kind Regards, Eddie Yam Alpha ESS UK

**Comment by DavidqStokes** [comment]
It's now 31st October and my battery charge / discharge setting still didn't work last night. Are others still having issues?

**Comment by dan22h** [comment]
I got fed up with the constant issues caused by the AlphaESS updates and have finally moved to MODBUS which is well worth the effort, the Inverter responds immediately to commands now and unlocks features that aren't possible otherwise, e.g. stopping the batteries charging from solar. 
Thanks for the hardwork carried out making this integration, it was a great help before Alpha ESS messed it up.

**Comment by Poshy163** [comment]
> Are others still having issues?

Personally not, what inverter you got? could be related some what to that? the PR i got going [here](https://github.com/CharlesGillanders/homeassistant-alphaESS/pull/135) should give more transparency regarding the currently set charge and discharge times (when the API polls) 

Should give you a little bit of info to give to alpha if its not fixed on your end

**Comment by braindeadmalc** [comment]
I still have the problem and have raised the case again as they requested

**Comment by Poshy163** [comment]
Hey there, do you have the full name (the model of it) so I can add it to the blacklist? Should show in your logs as Model: xxx Serial: xxxx

**Comment by tonbut** [comment]
Thank you everyone for updates, I'm not using this repo but read it to get the good information! Instead I'm using the [alphaess-openAPI](https://github.com/CharlesGillanders/alphaess-openAPI). I have a Smile5 and am having the same issues - reading works, but updates `updateChargeConfigInfo` isn't having any effect despite returning no errors. I hope alpha get their stuff sorted and at a minimum are more transparent about whats going on.

**Comment by pjhum** [comment]
Mine is Alpha Smile B3 serial AL7002022040171

**Comment by Poshy163** [comment]
Sorry @pjhum should of been more specific. when the integration is started, there is a string printed to the logs, this includes the model and serial number (these are case sensitive). it looks something like this: 

![image](https://github.com/user-attachments/assets/04b770dc-233f-41bf-b97b-c3b5b3d16cd7)

Cheers!

**Comment by pjhum** [comment]
Apologies, I'm fairly new to HA. which log, exactly? I've searched <settings> <system> <logs> and there is no entry adding an inverter.

Peter

**Comment by pjhum** [comment]
OK- removed the integration and re-added. The start up message was 
Alpha ESS Energy Statistics : AL7002022040171
Storion-SMILE-B3 (AlphaESS) 
that looks like the same string ?

My version still reports 0.5.5 BTW
Peter

**Comment by Poshy163** [comment]
To view logs in Home Assistant, you can navigate to Settings > System > Logs then click on view all logs.

If you reboot the integration, click ctrl + f (page finder thing) and look for "Model". It should bring up that line I posted earlier

**Comment by Poshy163** [comment]
Also, to force upgrade to 5.6.0, 

Under the integrations page on HACS, you can click the RELOAD DATA button under the SETTINGS tab (should be like the top right of the page) and then an update will appear.

Maybe do this and verify you can't set the discharge/charge via the buttons before i blacklist that model inverter

**Comment by braindeadmalc** [comment]
> Hey there, do you have the full name (the model of it) so I can add it to the blacklist? Should show in your logs as Model: xxx Serial: xxxx

Mine's "New Inverter: Serial: AL4001020050581, Model: Storion-SMILE-B3"

**Comment by pjhum** [comment]
Thanks, I now have 0.5.6 showing.The new controls have appeared.They do not appear to work, though. I pressed the "reset charge discharge button" and my AlphaEss app reports no change in settings.I'm presuming it should be instant, or nearly so?What I don't understand if the hardware is incompatible is that it was happily letting me set the overnight charge% until the last Alpha update/outage. PeterSent from Android deviceOn 2 Nov 2024 16:34, Joshua Leaper ***@***.***> wrote:
Also, to force upgrade to 5.6.0,
Under the integrations page on HACS, you can click the RELOAD DATA button under the SETTINGS tab (should be like the top right of the page) and then an update will appear.
Maybe do this and verify you can't set the discharge/charge via the buttons before i blacklist that model inverter

**Comment by Poshy163** [comment]
Yeah, if you click on any of the time periods, you then should be able to open your app and see the time settings set. Cheers for that all!

**Comment by Poshy163** [comment]
Cheers for the logs, ill add the inveter to the blacklist and push a minor update soon, the current/discharge charging time (1 & 2), and the charging range within the diagnostics section of the stuff will stay, just your ability to set anything (within the Configuration area) will be removed

**Comment by braindeadmalc** [comment]
> Cheers for the logs, ill add the inverter to the blacklist and push a minor update soon, the current/discharge charging time (1 & 2), and the charging range within the diagnostics section of the stuff will stay, just your ability to set anything (within the Configuration area) will be removed

Ah, I have a support ticket open with Alpha. If they fix the API issue will I be able to check it works?

**Comment by Poshy163** [comment]
ill keep it as a draft PR until you hear back, ive also talking with one of the guys behind the API to see if its intended or not.

Do keep us updated 👍🏻

**Comment by pjhum** [comment]
Thanks; I really need Alpha to fix this otherwise I am going to have to change energy tariffs! 

Resetting the charge/discharge data manually in the app makes the unit very little useto me.

I'm clinging to the belief that they broke this accidentally in the last upgrade rather than deprecating the older inverters without notice...though neither is a recommendation of the company.

Many thanks for all the hard work on this integration- it had served me very well.

Peter

**Comment by Poshy163** [comment]
All good @pjhum, just curious to see if this works, if set the charge/discharge data manually in the app (for some time) does it show up in the integration in the config area (you may need to restart the integration, or wait a few mins)

**Comment by pjhum** [comment]
Yes, I can see the charge/discharge settings and they updated pretty quickly (2-3 min max) in the 'diagnostic' card of the integration to match a change made in the app. I can't tell what _batHighCap or _batUseCap refer to in the 'config' secrtion, but they don't change. I had assumed they were to do with battery protection in the background. 


Peter

**Comment by Poshy163** [comment]
Here is from the API docs: 
batUseCap | Discharging Cutoff SOC [%]
batHighCap | Charging Stops at SOC [%]

Its basically the "bounds" for how much the batteries connected can charge and discharge, this is sent to the API when one of the buttons are pressed, obv if the set discharge and charge calls arnt working, i dont think it would be doing much

**Comment by pjhum** [comment]
OK- so it should set the same values that are reported in the diagnostic card. That is the part that isn't happening...

Peter

**Comment by Poshy163** [comment]
Well, yeah. which likely means that either you haven't pressed the charge/discharge (and waited up to a min for the next fetching of the API), orrr the setcharge and setdischarge isn't working, which seems to be the case here, i have asked for clarification by am yet to hear back

**Comment by Poshy163** [comment]
The batUseCap *should* also be the minimum amount the battery can discharge (its set as the Discharging cut off SOC in the settings) 

while the batHighCap is only shown as the Charging stops at SOC setting (aka is only used for force charging the battery)

**Comment by pjhum** [comment]
Yes, that is clear. The different names in the integration were confusing me, but a moment's thought tells me they need to be unique... 

I previously used the  Alpha ESS Set Battery Charge successfully in my automations to set summer night charging based on Solcast forecasts.

I was trying to set up & test an automation to stop my battery emptying into the car on Intelligent Octopus using the Alpha ESS Set Battery Discharge call when I realized neither was working... TBH the 'new' controls are less important to me than the existing ones resuming.

I have reached out to Alpha UK- without holding out much hope.

thanks again for your work and explanations.
Peter

**Comment by ppootle** [comment]
I have a very similar issue: Setting the charge and discharge times worked perfectly via the integration until about 2 weeks ago, but no longer seems to work. Only reading the diagnostics now seems to work via the integration. Manually setting the charge and discharge time via the app still works - and then shows correctly in the diagnostics. Have removed and re-added the integration but can't get it to work to set the timings anymore.

**Comment by Poshy163** [comment]
@ppootle do you also have a Storion-SMILE-B3?

**Comment by Poshy163** [comment]
@pjhum you might have some luck with alpha support, 
![image](https://github.com/user-attachments/assets/15897fab-88c0-4ce1-acea-6c8689aa195f)
translates to: "This charge and discharge setting is applicable to all models. If it does not work, you need to submit a customer complaint and contact our after-sales service."

**Comment by ppootle** [comment]
No. It's a Smile5INV.

**Comment by pjhum** [comment]
Thanks @Poshy163, I'll wait and see when they get back to me.

Was that a Google translate job or can you read Chinese? If so, I'm impressed...

**Comment by renamedthewild** [comment]
I cant get the discharge setting to work with t10HV or Smile5

**Comment by Poshy163** [comment]
Haha, I used google translate, I'm not that good.

Also, did some more digging. Seems to be a widespread issue. See the video below. Setting the discharge time seems to disable it? I'll do some digging as to if it was something on my end. Or the API


https://github.com/user-attachments/assets/7d74ccd6-8f63-4157-a331-7907c3e6bfa4

**Comment by Poshy163** [comment]
Ah well, found the issue. if anyone needs an fix now, and knows how to get to the source files of home assistant,  

On line 93 of the coordinator.py, there is a 0 in the updateDisChargeConfigInfo call, change to a 1, save. then reboot and it will work. otherwise and update will go out as soon as Charles approves my PR

**Comment by renamedthewild** [comment]
Can this fix be done at a home asssit level ?

Sent from Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by Poshy163** [comment]
you can use the visual studio code add-on, but i would not recommend it if you are not sure, rather wait for the PR to be merged

**Comment by renamedthewild** [comment]
Got it

Sent from Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by renamedthewild** [comment]
> Ah well, found the issue. if anyone needs an fix now, and knows how to get to the source files of home assistant,
> 
> On line 93 of the coordinator.py, there is a 0 in the updateDisChargeConfigInfo call, change to a 1, save. then reboot and it will work. otherwise and update will go out as soon as Charles approves my PR

I made the change, however it did not seem to resolve the issue

**Comment by pjhum** [comment]
Cheers- sheer coincidence that I only started to try and control discharge after the sun had gone in for the year in Lancashire, so didn't see the link! 

I'll wait for the merge- I'm not a PYTHON fan...

Thanks again for all your effort on this.

Peter

**Comment by Poshy163** [comment]
> I made the change, however it did not seem to resolve the issue
assuming you have restarted home assistant after making the change, and waiting 10 mins (since the last press as there is a cooldown implemented by alphaess)

Does anything show in the app? the fix worked for me, and i can show a video verifying if you'd like?

i would just recommend waiting for the patch and then giving it a go

**Comment by Poshy163** [comment]
Interesting. i would wait for the .1 release, and update. if this still occurs you can make another issue. it could be case of you have changed the wrong variable.

**Comment by DavidqStokes** [comment]
I am a little unclear here about the scope of this fix.
Does this fix the new force charge/discharge feature only, or does it also fix the wider issue of the failure to set charge/discharge parameters via automations?

**Comment by Poshy163** [comment]
> Does this fix the new force charge/discharge feature only

It only fixes the features offered by this integration, so if there is any automations tied to it (that uses this). Not sure what external providers like Amber use tho

**Comment by DavidqStokes** [comment]
I'll check what happens when my automations run (trigger the actions to make the set charge / discharge) this evening

**Comment by Poshy163** [comment]
The charge should work, the discharge wont until the latest PR is merged

**Comment by pjhum** [comment]
I’ve had a response from Alpha UK- they suggest trying using a different URL for the API-https://apiservice-eu.alphaess.com/v1/Open/XXXXX

I don’t know which URL the integration uses. If it isn’t this I need to find some tools to interrogate the API manually, I gues…

Peter

**Comment by ppootle** [comment]
I got this response from UK support: 

"Could you please update the URL and API address to below and try again?
European Center URL: https://apicloud-eu.alphaess.com/
European Center API address: https://apiservice-eu.alphaess.com/v1/Open/XXXXX
"

Not sure what i need to do with it though. I get user "user does not exist" if i try the URL.

**Comment by pjhum** [comment]
You need an API test tool- Postman or whatever.

I've downloaded SoapUI which is free but I'm struggling to frame a request (any request) to test it.  The API documentation is just that, not a tutorial... plus I've never used the software before. 

I'm on some other training tomorrow but will have a dig later in the week if I get time- though for it to help in HA the source code for the integration would have to be updated anyway. Thinking about it, that might be the easiest test path.

**Comment by Poshy163** [comment]
The update has been pushed, so you can force the  fetch the new update by going onto the integration (on the HACS page), click on the three dots (⋮) on the integration’s card and then update information

**Comment by Poshy163** [comment]
> "Could you please update the URL and API address to below and try again? European Center URL: https://apicloud-eu.alphaess.com/ European Center API address: https://apiservice-eu.alphaess.com/v1/Open/XXXXX "

Interesting, we dont keep the baseURL (aka the URL to the API) anywhere in this repo, its kept in the openAPI repo [here](https://github.com/CharlesGillanders/alphaess-openAPI)

Changing that as the end user is not easy, and i would recommend if you want to test it, you use something like postman (there is a example collection in that openAPI repo) 

Maybe try the .1 release that just got put out and see if that fixes it. otherwise might need to re-work where the base URL is deprived from.

**Comment by pjhum** [comment]
Explains why I couldn't find the base URL in the repository files. I'll have a go at trying to query the 'new' API if the update brings no joy. Peter Sent from Android deviceOn 4 Nov 2024 23:12, Joshua Leaper ***@***.***> wrote:

"Could you please update the URL and API address to below and try again? European Center URL: https://apicloud-eu.alphaess.com/ European Center API address: https://apiservice-eu.alphaess.com/v1/Open/XXXXX "

Interesting, we dont keep the baseURL (aka the URL to the API) anywhere in this repo, its kept in the openAPI repo here
Changing that as the end user is not easy, and i would recommend if you want to test it, you use something like postman (there is a example collection in that openAPI repo)
Maybe try the .1 release that just got put out and see if that fixes it. otherwise might need to re-work where the base URL is deprived from.

**Comment by braindeadmalc** [comment]
> The update has been pushed, so you can force the fetch the new update by going onto the integration (on the HACS page), click on the three dots (⋮) on the integration’s card and then update information

Hi, I've tried the update on my B3 and no joy I'm afraid.

Also, for info, I've had no update on my case from Alpha recently.

**Comment by renamedthewild** [comment]
Still no good for me setting 1 hour discharge just sets a 1hour window the battery can be drawn from the load … it not dispatching

Sent from Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by pjhum** [comment]
Same with me; Storion Smile B3 is not working on charge or discharge setting even with the .1 update.
Peter

**Comment by Poshy163** [comment]
Just curious, if you click it, then open the app (fresh open the app) then have a look at the dis-charge settings, does it show there?

**Comment by pjhum** [comment]
No, the app settings reflect whatever I last set in the Alpha app. I suspect it is an Alpha issue. I need to get my head round framing a suitable HTTP request to test the alternative URL, but I'm sitting through seven hours of CPC training today

Peter

**Comment by renamedthewild** [comment]
so what is the expected outcome when i push discharge 60mins now ?

**Comment by renamedthewild** [comment]
![image](https://github.com/user-attachments/assets/c0e93a43-1e35-44f3-818d-a2484a5ead63)

This is all I get when I press the button, however this is not dispatching this is just discharge timed control..

Can this API dispatch ?

**Comment by Poshy163** [comment]
That is the expected outcome for your inverter it seems, per the API: 
> According to SN Set discharge information，Setting frequency 24 hours, set once a day

**Comment by pjhum** [comment]
I read that it had previously been lifted to once in 10 minutes?

**Comment by Poshy163** [comment]
Yeah, it was reduced for being able to set it every 10 mins, clearly the API docs weren't updated 🙄

**Comment by Poshy163** [comment]
> Oh ok - sorry assumed open API could configure battery dispatching - so you could discharge your battery to the grid during high prices 

Honestly that's what I thought, but different batteries may behave differently

**Comment by renamedthewild** [comment]
oh well back to modbus it is

**Comment by renamedthewild** [comment]
I also use my self a SMILE-T10-HV-INV - that has modbus over TCP and that works amazing

**Comment by pjhum** [comment]
The original B3 seems a bit of an outlier. did you have to mess about to get the converter working? mine is set up as a simple TCP server with the serial side on 9600 no parity as per the docs I can find.  HA has RTU over TCP set, but getting nothing.  

My freebie modbus tester connects to the bridge just gets time outs. 

Now thinking the B3 doesn't do modbus at all...

**Comment by DavidqStokes** [comment]
I have now downloaded the new 5.6.1 release and will see whether changing the charge/discharge settings via automation (service call) works this evening with my SMILE B3.

**Comment by braindeadmalc** [comment]
I just got another response to my support ticket from Alpha. I;m not quite sure where this leaves us,

"Sorry for our late reply. According to the Web API team, there are no reported issues with the OpenAPI command integration at this time. Could you please try again?
For questions related to OpenAPI, we recommend reaching out on the GitHub forum, where additional support is available. Thank you for your understanding"

**Comment by pjhum** [comment]
Looking on the German users' forum, it seems European data is now stored on the (new) European server. I suspect this is also hosting the new API that UK customer support gave me.  (https://apiservice-eu.alphaess.com/v1/Open/XXXXX)

**Comment by purrcat33a** [comment]
I'd been holding out for a reply from Alpha regarding this same issue on my one year old B3-plus. I got exactly the same response from them this morning as braindeadmalc. 

Like others , I'd been able to set charge and discharge periods and desired SOC using the OpenAPI via Home Assistant a couple of times a day so as to make best use of Octopus Agile and prevailing weather conditions. Since the global update I can read SOC but cannot set it or charge periods via HA. I pulled the latest version of the OpenAPI yesterday. I believe I am on 0.5.6.1 version.

**Comment by Poshy163** [comment]
Interesting, seems like ill have to make the API dynamic, and what will happen is you guys will have to manually edit your files and place the API URL in the files (will require an update to the integration + API)

**Comment by CharlesGillanders** [comment]
> Interesting, seems like i'll have to make the API dynamic, and what will happen is you guys will have to manually edit your files and place the API URL in the files (will require an update to the integration + API) 

The optimum solution would be to have the API library be aware of both API endpoints and to accept a parameter that would select a specific endpoint for each time data is requested.

We would then be able to configure the Home Assistant integration with an extended setup dialogue that asked for a radio button type question "EU or worldwide API endpoint" 

I can make the appropriate change to the library later this week.

**Comment by Poshy163** [comment]
Wonderful, ill see if i can get a full list of region specific list of URL's, tho not holding out for much hope

**Comment by DavidqStokes** [comment]
I'm in the UK and despite downloading 5.6.1 yesterday my automations still failed to effect any changes to the charge / discharge settings on of my SMILE B3 last night.
It does however sound like you have a handle on this - thanks as always for the excellent work on this integration.

**Comment by tonbut** [comment]
I've just been experimenting with the new EU URL https://apiservice-eu.alphaess.com/v1/Open/getLastPowerData . It seems to almost work, but every call fails with the error `{'code': 103, 'info': 'Timestamp is Wrong！'}`  Switching back to the old URL, it works again. Not sure if Alpha has updated the API in other ways too? I havent seen any doc updates.

**Comment by Poshy163** [comment]
Im unfortunately not in the EU, so i wont be much help, but is there any documentation on this website: https://apicloud-eu.alphaess.com/#/?

**Comment by tonbut** [comment]
I wish Alpha made this information easier to find! Does anyone know how to get login credentials for  https://apicloud-eu.alphaess.com/#/ ? If I use the ones created on https://open.alphaess.com/login it says "no such user". 

@pjhum on the original API the timestamp was in seconds since 1970 Epoch which should be timezone independent AFAIK

**Comment by pjhum** [comment]
I've been wrong before...😉I was using the timestamp generator linked from the open API documentation, and it appeared to return a value in Beijing local time when I decoded it.May simply be the decoding that was corrected to their local time- Google wasn't making such a good job of translating the page from Chinese.I couldn't log in to the German server's API either, though I'm not confident my request was properly formed.PeterSent from Android deviceOn 7 Nov 2024 10:06, Tony Butterfield ***@***.***> wrote:
I wish Alpha made this information easier to find! Does anyone know how to get login credentials for  https://apicloud-eu.alphaess.com/#/ ? If I use the ones created on https://open.alphaess.com/login it says "no such user".
@pjhum on the original API the timestamp was in seconds since 1970 Epoch which should be timezone independent AFAIK

**Comment by CharlesGillanders** [comment]
I am beginning to think that there is confusion within AlphaESS between the OpenAPI and their older commercial API services.

According to this page [https://urlscan.io/domain/alphaess.com](https://urlscan.io/domain/alphaess.com) the apiservice.alphaess.com URL has been around since 2021, whereas the URL for openapi.alphaess.com is only from 2023.

In the absence of any comment on the OpenAPI developer portal I wonder if there's a chance that people are being given mistaken information by the EU support team?

**Comment by braindeadmalc** [comment]
I may be wrong but something got fixed!
My automations seem to be working again.
I've tried a few tests and I seem to be able to set the charging period via HA again.

**Comment by purrcat33a** [comment]
I'm seeing the same thing as braindeadmalc. It does seem to be working.

I did, initially, find that my two automations were no longer valid. I pulled the latest GitHub version and it made no difference. So I investigated further and discovered that the SOC condition definition has changed from (whatever it was before, as I didn't write it down) to "current <serial number"_State of battery level .

I've run both automations a number of times and it appears to be setting the battery charge/discharge settings properly and quickly.

Fingers crossed !

Who ever has changed things to make it work .... thank you!

**Comment by pjhum** [comment]
My original automations run overnight- I'll see if they've restarted working.If so, I'll then see if the new force charge/discharge features work on the B3.PeterSent from Android deviceOn 8 Nov 2024 18:08, purrcat33a ***@***.***> wrote:
I'm seeing the same thing as braindeadmalc. It does seem to be working.
I did, initially, find that my two automations were no longer valid. I pulled the latest GitHub version and it made no difference. So I investigated further and discovered that the SOC condition definition has changed from (whatever it was before, as I didn't write it down) to "current <serial number"_State of battery level .
I've run both automations a number of times and it appears to be setting the battery charge/discharge settings properly and quickly.
Fingers crossed !
Who ever has changed things to make it work .... thank you!

**Comment by DavidqStokes** [comment]
I'm also in the UK with a SMILE B3 battery, and my overnight automation was able to set the charge / discharge parameters last night for the first time since the AlphaESS upgrade.

I'm a bit lost as to what the problem actually was, and who did what to fix it, but my thanks to everyone involved in getting to the bottom of this and getting the issue fixed.

**Comment by pjhum** [comment]
+1 here with another B3

Automations worked, and the “reset charge/discharge” button reset my charge parameters to the values set in the boxes and cleared the charge times I had set.

I’ll experiment with forced charge and discharge buttons later on, but something has definitely improved.

Thanks to the integration developers for the effort this has taken- even if ultimately it turns out to be Alpha’s SNAFU

**Comment by Poshy163** [comment]
Haha what an ending, must of been something wrong with the firmware on the batteries or something, super weird that it just *suddenly* started working one day 

Maybe enough people complained that it finally got escalated 🙈

**Comment by pjhum** [comment]
OK, back to what My B3 will and won't do.Set charge works fine, both as an action in an automation and by the new x-minute charge buttons. Big relief for me as I use Solcast forecasting.At present, I cannot alter the discharge settings via this integration- though If someone at Alpha is working in the background that might change? If I alter the discharge parameters in the app, it is reflected in this integration.Will report back if anything changes.Meanwhile I expect I can solve my discharge problems by forcing the unit to charge instead- I'm not on an export tariff (on FIT here ) so it isn't critical for me to force export, just stop the battery emptying during cheap periods.PeterSent from Android deviceOn 9 Nov 2024 10:55, Joshua Leaper ***@***.***> wrote:
Haha what an ending, must of been something wrong with the firmware on the batteries or something, super weird that it just suddenly started working one day
Maybe enough people complained that it finally got escalated 🙈

**Comment by purrcat33a** [comment]
I asked Alpha what they did to correct the problem (if, indeed, it was them).

The reply that I got was this : "There's a bug in OpenAPI identified and is expected to be fixed this week, therefore you may still experience the issue in coming days. Sorry for any inconvenience caused."

I'm not sure exactly what this is saying, but it does suggest that something has been found and is being addressed. It doesn't explain why things magically started working last week though.

**Comment by DavidqStokes** [comment]
Sounds like AlphaASS are one of the first to use quantum computing at scale. Rather than relying on logic to produce known outcome based on inputs, the result is actually the outcome of some inexplicable probability function...

**Comment by purrcat33a** [comment]
> Sounds like AlphaASS are one of the first to use quantum computing at scale. Rather than relying on logic to produce known outcome based on inputs, the result is actually the outcome of some inexplicable probability function...

They probably hitched a ride on the infinite improbability drive and derived a result from absolutely everywhere.

**Comment by pjhum** [comment]
As of this morning, I see discharge control also working via this integration on my B3.

AFAIK my B3 now works fully with this, and the model does not need blacklisting.

I am thinking this was a bug in the API all along.

As an aside, now I have a different modbus gateway the Smile B3 model also supports modbus control.

Thanks again to the developers of this integration.

**Comment by frostilm** [comment]
It's end of 2025, and we're still having this problem. 

Right after we have installed the system (a few days ago), we noticed we couldn't update the charging cutoff SOC. Initially, discharging cutoff SOC and time rules can be set in the app. Today, we've tried the web client and it started saying network busy on the charge/discharge setting page only, as well as in the app. Now both charging / discharging cutoff SOC are not working. 

I'm suspecting this is some kind of measure from AlphaESS disabling people from having longer lifespan on their batteries. 

Model: SMILE-M5-S-INV

https://github.com/user-attachments/assets/e611284e-3050-4155-a98a-066db602dfe8
